streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -255,6 +255,15 @@ private void restorePartition(final ConsumerRecords<byte[], byte[]> allRecords,
                 throw new TaskMigratedException(task, topicPartition, endOffset, pos);
             }
 
+            // not a KTable changelog topic but state store
```

Comment:
We need the check for `changelog topic`, right? (for input topics the check would not be applicable)